tequilapi/endpoints/eth_info.go

```diff
@@ -0,0 +1,66 @@
+/*
+ * Copyright (C) 2018 The "MysteriumNetwork/node" Authors.
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+package endpoints
+
+import (
+	"net/http"
+
+	"github.com/julienschmidt/httprouter"
+	"github.com/mysteriumnetwork/node/tequilapi/contract"
+	"github.com/mysteriumnetwork/node/tequilapi/utils"
+)
+
+type l2EthClientURLProvider interface {
+	CurrentClientOrder() []string
+}
+
+type ethInfoEndpoint struct {
+	ethL2URLProvider l2EthClientURLProvider
+}
+
+func newEthInfoEndpoint(ethClient l2EthClientURLProvider) *ethInfoEndpoint {
+	return &ethInfoEndpoint{ethL2URLProvider: ethClient}
+}
+
+// EthInfo provides info about active eth client
+// swagger:operation GET /eth-info
+// ---
+// summary: ETH Client Info
+// description: Provides active info about eth client
+// response:
+//	200:
+//		schema:
+//			"$ref": "#/definitions/EthInfoResponse"
+func (e ethInfoEndpoint) EthInfo(resp http.ResponseWriter, _ *http.Request, _ httprouter.Params) {
+	var response contract.EthInfoResponse
+	clientURLOrder := e.ethL2URLProvider.CurrentClientOrder()
+	if len(clientURLOrder) != 0 {
```

Comment:
It goes like this:
* go through the list of rpc clients, use them all in current order
  * if one doesn't respond push it to the end of the list
  * if one responds stop and return result

So I would honestly return the whole list and explain they're used in order, as it's possible to use 2nd node in the list if the 1st didnt respond. Unless you think that would just confuse the end user.